### ISSUE TITLE
Modules brokes multiple pipeline

### ISSUE THREAD

**Issue by ph** [elastic/logstash#7369]
With the merge of [modules](https://github.com/elastic/logstash/pull/7284) we sadly broke our implementation of multiple pipelines.

If you define your pipeline in the  **pipelines.yml** and your start Logstash you will get:

```
[2017-06-07T14:13:57,503][INFO ][logstash.setting.writabledirectory] Creating directory {:setting=>"path.dead_letter_queue", :path=>"/Users/ph/es/logstash/data/dead_letter_queue"}
ERROR: No configuration file was specified. Perhaps you forgot to provide the '-f yourlogstash.conf' flag?
usage:
  bin/logstash -f CONFIG_PATH [-t] [-r] [] [-w COUNT] [-l LOG]
  bin/logstash --modules MODULE_NAME [-M "MODULE_NAME.var.PLUGIN_TYPE.PLUGIN_NAME.VARIABLE_NAME=VALUE"] [-t] [-w COUNT] [-l LOG]
  bin/logstash -e CONFIG_STR [-t] [--log.level fatal|error|warn|info|debug|trace] [-w COUNT] [-l LOG]
  bin/logstash -i SHELL [--log.level fatal|error|warn|info|debug|trace]
  bin/logstash -V [--log.level fatal|error|warn|info|debug|trace]
```

I've track it down to this part of the [code](https://github.com/elastic/logstash/blob/master/logstash-core/lib/logstash/bootstrap_check/default_config.rb#L73-L75):

```ruby
      if !any_config?
        raise LogStash::BootstrapCheckError, I18n.t("logstash.runner.missing-configuration")
      end
```

If we look at the previous commit, this is a lot different.

```ruby
if settings.get("config.reload.automatic") && settings.get_setting("config.string").set?
   LogStash::BootstrapCheckError, I18n.t("logstash.runner.reload-with-config-string")
end
```

We will to rethink our check implementation and add integration test for the multiple pipeline.

**Comment by ph** [comment]
cc @untergeek @suyograo

**Comment by guyboertje** [comment]
I am working on this - solution v soon.

**Comment by ph** [comment]
This was merged in master #7452
